.travis.yml

```diff
@@ -22,6 +22,11 @@ notifications:
     on_success: never
     on_failure: always
 
+before_script:
+  - 'if [ "$IMMUTABLE_VERSION" ]; then npm install immutable@$IMMUTABLE_VERSION; fi'
+env:
+  - IMMUTABLE_VERSION=3
```

Comment:
I think for v3 it's reasonable to rely on whatever the `package.json` file contains, so maybe just `IMMUTABLE_VERSION=`?
When stable v4 is released, and `package.json` refers to it, then v3 will be the alternate path we can maintain in `.travis.yml`.

Reply:
Actually, since this adds a significant number of builds to the Travis configuration, why about replacing the whole `env` block with a [single `include` entry](https://docs.travis-ci.com/user/build-matrix/#explicitly-including-jobs), that would e.g. run against Immutable v4 (via an `env`), a single `os` (I'd go with `linux`, which is the default so can be omitted I believe), and a single version of Node.js (any LTS would work, maybe v10? Not opposed to v8 too, or even v6 as the minimum supported version)?

Reply:
Let me know if I'm being unclear and I will give it a shot in your branch.